### PR TITLE
Generic SAML Stepup Provider Integration

### PR DESCRIPTION
**Pull request by DRvanR**
- [x] Translations

### CODE REVIEW

File: src/Surfnet/StepupGateway/GatewayBundle/Saml/Proxy/ProxyStateHandler.php [hunk 1]
```diff
@@ -27,7 +27,7 @@ class ProxyStateHandler
     /**
      * @var \Symfony\Component\HttpFoundation\Session\SessionInterface
      */
-    private $session;
+    protected $session;
```
Comment (rjkip) [inline src/Surfnet/StepupGateway/GatewayBundle/Saml/Proxy/ProxyStateHandler.php]: Is there a specific reason for exposing this property to subclasses at this point?

Comment (DRvanR) [inline src/Surfnet/StepupGateway/GatewayBundle/Saml/Proxy/ProxyStateHandler.php]: There was, there no longer is :wink:

File: src/Surfnet/StepupGateway/SamlStepupProviderBundle/Provider/ConnectedServiceProviders.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ */
+
+namespace Surfnet\StepupGateway\SamlStepupProviderBundle\Provider;
+
+use Surfnet\StepupGateway\GatewayBundle\Service\SamlEntityService;
+use Surfnet\StepupGateway\SamlStepupProviderBundle\Exception\InvalidArgumentException;
+use Surfnet\StepupGateway\SamlStepupProviderBundle\Exception\UnknownProviderException;
+
+/**
+ * @todo discuss (im)mutability
```
Comment (rjkip) [inline src/Surfnet/StepupGateway/SamlStepupProviderBundle/Provider/ConnectedServiceProviders.php]: If there's no need for runtime connecting of service providers, I suppose we should opt for constructor injection.

File: src/Surfnet/StepupGateway/SamlStepupProviderBundle/DependencyInjection/Compiler/StateHandlerSessionPass.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * {@inheritdoc} This is required to ensure that our NamespacedAttributeBag is registered in the session handler
+     * before the session is started.
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $container
+            ->getDefinition('session')
+            ->addMethodCall('registerBag', [new Reference('gssp.session.namespaced_attribute_bag')])
+            ->addMethodCall('start');
```
Comment (rjkip) [inline src/Surfnet/StepupGateway/SamlStepupProviderBundle/DependencyInjection/Compiler/StateHandlerSessionPass.php]: This OK?

**Comment by rjkip** [comment]
[Comme ça, monsieur?](https://github.com/SURFnet/Stepup-Gateway/commit/be09b01d74d319e5f18fb3444a997ec08b470157)

**Comment by rjkip** [comment]
Sup @DRvanR

**Comment by DRvanR** [comment]
@rjkip  You know, working on it a bit and all that... :wink:

**Comment by rjkip** [comment]
Are there still things to be done here?

**Comment by DRvanR** [comment]
Second Factor verification for when the Tiqr SF is being used to log in to a SP with LoA > 2, gimme a minute.... - mostly testing atm

**Comment by rjkip** [comment]
As far as I can see :+1:

**Pull request merged**